jdk10/Dockerfile

```diff
@@ -3,86 +3,86 @@ FROM openjdk:10-jdk
-	&& ln --symbolic "${GROOVY_HOME}/bin/groovyc" /usr/bin/groovyc \
-	&& ln --symbolic "${GROOVY_HOME}/bin/groovyConsole" /usr/bin/groovyConsole \
-	&& ln --symbolic "${GROOVY_HOME}/bin/groovydoc" /usr/bin/groovydoc \
-	&& ln --symbolic "${GROOVY_HOME}/bin/groovysh" /usr/bin/groovysh \
-	&& ln --symbolic "${GROOVY_HOME}/bin/java2groovy" /usr/bin/java2groovy \
-	\
-	&& echo "Editing startGroovy to include java.xml.bind module" \
-	&& sed -i 's|startGroovy ( ) {|startGroovy ( ) {\n    JAVA_OPTS="$JAVA_OPTS --add-modules=ALL-SYSTEM"|' "${GROOVY_HOME}/bin/startGroovy" \
-	\
-	&& echo "Cleaning up build dependencies" \
-	&& echo $(apt-mark showauto) \
-	&& apt-get remove --yes --purge \
-		dirmngr \
-		gnupg \
-	&& apt-get autoremove --yes --purge \
-	\
-	&& echo "Adding groovy user and group" \
-	&& groupadd --system --gid 1000 groovy \
-	&& useradd --system --gid groovy --uid 1000 --shell /bin/bash --create-home groovy \
-	&& mkdir --parents /home/groovy/.groovy/grapes \
-	&& chown --recursive groovy:groovy /home/groovy \
-	\
-	&& echo "Symlinking root .groovy to groovy .groovy" \
-	&& ln -s /home/groovy/.groovy /root/.groovy
+    && echo "Downloading Groovy" \
+    && wget --no-verbose --output-document=groovy.zip "https://dist.apache.org/repos/dist/release/groovy/${GROOVY_VERSION}/distribution/apache-groovy-binary-${GROOVY_VERSION}.zip" \
+    \
+    && echo "Installing build dependencies" \
+    && apt-get update \
+    && apt-get update && apt-get install --yes --no-install-recommends \
+        dirmngr \
+        gnupg \
+    && rm --recursive --force /var/lib/apt/lists/* \
+    \
+    && echo "Importing keys listed in http://www.apache.org/dist/groovy/KEYS from key server" \
+    && export GNUPGHOME="$(mktemp -d)" \
+    && for key in \
+        "7FAA0F2206DE228F0DB01AD741321490758AAD6F" \
+        "331224E1D7BE883D16E8A685825C06C827AF6B66" \
+        "34441E504A937F43EB0DAEF96A65176A0FB1CD0B" \
+        "9A810E3B766E089FFB27C70F11B595CEDC4AEBB5" \
+        "81CABC23EECA0790E8989B361FF96E10F0E13706" \
+    ; do \
+        for server in \
+            "ha.pool.sks-keyservers.net" \
+            "hkp://p80.pool.sks-keyservers.net:80" \
+            "pgp.mit.edu" \
+        ; do \
+            echo "  Trying ${server}"; \
+            if gpg --keyserver "${server}" --recv-keys "${key}"; then \
+                break; \
+            fi; \
+        done; \
+    done \
+    && if [ $(gpg --list-keys | grep -c "pub ") -ne 5 ]; then \
+        echo "ERROR: Failed to fetch GPG keys" >&2; \
+        exit 1; \
+    fi \
+    \
+    && echo "Checking download signature" \
+    && wget --no-verbose --output-document=groovy.zip.asc "https://dist.apache.org/repos/dist/release/groovy/${GROOVY_VERSION}/distribution/apache-groovy-binary-${GROOVY_VERSION}.zip.asc" \
+    && gpg --batch --verify groovy.zip.asc groovy.zip \
+    && rm --recursive --force "${GNUPGHOME}" \
+    && rm groovy.zip.asc
+RUN set -o errexit -o nounset \
```

Comment:
This is a problem -- you've split across two `RUN` instructions, so the `groovy.zip` you remove a few lines later won't actually be removed properly (it'll become a layer whiteout instead).

cc @yosifkit

Reply:
Ah, that makes sense. Good catch.  Working on a fix.

Reply:
Here's what I'm thinking: https://github.com/groovy/docker-groovy/pull/37